dss/events/__init__.py

```diff
@@ -52,23 +46,25 @@ def get_deleted_bundle_metadata_document(replica: Replica, key: str) -> dict:
 @lru_cache(maxsize=2)
 def record_event_for_bundle(replica: Replica,
                             key: str,
-                            flashflood_prefix: str=None,
+                            flashflood_prefixes: typing.Tuple[str, ...]=None,
                             use_version_for_timestamp: bool=False) -> dict:
     """
     Build the bundle metadata document, record it into flashflood, and return it
     """
     # TODO: Add support for unversioned tombstones
     fqid = key.split("/", 1)[1]
-    ff = FlashFlood(resources.s3,  # type: ignore
-                    Config.get_flashflood_bucket(),
-                    flashflood_prefix or replica.flashflood_prefix)
+    if flashflood_prefixes is None:
+        flashflood_prefixes = replica.flashflood_prefix_write
     metadata_document = _build_bundle_metadata_document(replica, key)
     if use_version_for_timestamp:
         _, version = fqid.split(".", 1)
         event_date = datetime_from_timestamp(version)
     else:
         event_date = datetime.utcnow()
-    ff.put(json.dumps(metadata_document).encode("utf-8"), event_id=fqid, date=event_date)
+    for pfx in flashflood_prefixes:
+        ff = FlashFlood(resources.s3, Config.get_flashflood_bucket(), pfx)
+        if not ff.event_exists(fqid):
+            ff.put(json.dumps(metadata_document).encode("utf-8"), event_id=fqid, date=event_date)
```

Comment:
Should there be a catch to ensure all prefixes are attempted to be written to? 
if one of the first prefixes in the list fails, i assume we want the later ones to still be written to?

Reply:
Merged before I saw this comment. But I don't know what the correct strategy is for that case.